app/controllers/stories_controller.rb

```diff
@@ -176,7 +176,7 @@ def handle_user_index
     redirect_if_view_param
     return if performed?
 
-    set_surrogate_key_header "articles-user-#{@user.id}"
+    set_surrogate_key_header @user.record_key
```

Comment:
What do you think about making this more flexible, we should at least add `@stories` record keys to this as this returns a collection of articles.

What about:

```ruby
set_surrogate_key_header @user.record_key, "articles", @stories.map(&:record_key)
```

?

The article purging code should get in the code if we merge https://github.com/thepracticaldev/dev.to/pull/5606